app/controllers/policy_classes_controller.rb

```diff
@@ -46,17 +45,11 @@ def create
   def show; end
 
   def update
-    new_policies = policies_params[:policies]
-
-    @policy_class.policies.each do |policy|
-      value = new_policies[policy["id"].to_s]
-
-      policy["status"] = value if value.present?
+    if @policy_class.update(policy_class_params)
+      redirect_to @planning_application, notice: "Successfully updated policy class"
+    else
+      render :show
```

Comment:
Not sure how the form ended up in `show` and not `edit`...but we're about to overhaul a lot of things in upcoming work and not sure how it will look, so leaving it that way for now!